tests/model_fields/tests.py

```diff
@@ -60,6 +66,207 @@ def test_field_str(self):
         self.assertEqual(force_str(f), 'model_fields.Foo.a')
 
 
+class ReadonlyTestsMixin(object):
+
+    @classmethod
+    def populate_database_fields(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def setUpClass(cls):
+        super(ReadonlyTestsMixin, cls).setUpClass()
+
+        cls.populate_database_fields()
+
+    def test_readonly_option(self):
+        # Default value
+        self.assertFalse(ModelWithReadonlyField._meta.get_field('not_readonly').readonly)
+
+        # When the readonly attribute is set
+        self.assertTrue(ModelWithReadonlyField._meta.get_field('readonly_int1').readonly)
+
+    def test_create(self):
+        with CaptureQueriesContext(connection=connection) as capture:
+            inst = ModelWithReadonlyField.objects.create(
+                not_readonly='not_readonly',
+                readonly_int1=3,
+            )
+
+        self.assertEqual(len(capture.captured_queries), 2)
+        query = capture.captured_queries[0]
+
+        self.assertIn('not_readonly', query['sql'])
+        self.assertNotIn('readonly_int1', query['sql'])
+        self.assertNotIn('readonly_int2', query['sql'])
+        self.assertEqual(inst.readonly_int1, 1)
+        self.assertEqual(inst.readonly_int2, 2)
+
+    def test_save_new(self):
+        with CaptureQueriesContext(connection=connection) as capture:
+            inst = ModelWithReadonlyField(
+                not_readonly='not_readonly',
+                readonly_int1=1,
+            )
+            inst.save()
+
+        self.assertEqual(len(capture.captured_queries), 2)
+        query = capture.captured_queries[0]
+
+        self.assertIn('not_readonly', query['sql'])
+        self.assertNotIn('readonly_int1', query['sql'])
+        self.assertNotIn('readonly_int2', query['sql'])
+        self.assertEqual(inst.readonly_int1, 1)
+        self.assertEqual(inst.readonly_int2, 2)
+
+    def test_save_update(self):
+        inst = ModelWithReadonlyField.objects.create(
+            not_readonly="not_readonly",
+            readonly_int1=5,
+        )
+        inst.not_readonly = 'foo'
+        inst.readonly_int1 = 42
+
+        with CaptureQueriesContext(connection=connection) as capture:
+            inst.save()
+
+        query = capture.captured_queries[0]
```

Comment:
I'm not sur one is needed here. It's present in the other tests when we need tu make sur we're checking the right query. That being said, I can add a check if you think it's better with than without.